deployment.yaml

```diff
@@ -0,0 +1,70 @@
+---
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: node-controller-sa
+  namespace: default
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: node-controller-role
+rules:
+  - apiGroups: [""]
+    resources: ["nodes"]
+    verbs: ["list", "watch"]
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["create", "patch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: node-controller-access
+subjects:
+  - kind: ServiceAccount
+    name: node-controller-sa
+    namespace: default
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: node-controller-role
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: node-controller
+  namespace: default
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: node-controller
+  template:
+    metadata:
+      labels:
+        app: node-controller
+    spec:
+      serviceAccountName: node-controller-sa
+      containers:
+        - name: node-controller
+          image: us-central1-docker.pkg.dev/peteryizhong-gke-dev/peteryizhong-repository/node-controller:0.1
+          imagePullPolicy: Always
+          env:
+          - name: SELECTOR
+            value: "cloud.google.com/gke-nodepool=default-pool"
+          - name: SECURITY_POLICY
+            value: "yi-edge-sp"
```

Comment:
ditto for secutiy policy, put something obvious here

Reply:
Done